### DISCUSSION TITLE
Select starting nodes class (for `multistart` decoding)

### DISCUSSION THREAD

**Discussion by fedebotu** [ai4co/rl4co#117] [General]
At the moment, we have a `select_start_nodes` ([here](https://github.com/ai4co/rl4co/blob/83b85500171972d204727f36a911fd9fde66f9a2/rl4co/utils/ops.py#L108)) that is not very modular (i.e. for each environment, you have to manually make an "if---else"). Also, it is not clear at all for newcomers. We should make this as a class such that a user can create their own and pass it. 
In the same class, we should also include a method to [get the number of starts](https://github.com/ai4co/rl4co/blob/83b85500171972d204727f36a911fd9fde66f9a2/rl4co/utils/ops.py#L96) for the environment (e.g. for PDP, we have half of the nodes as possible starting points)

Related: See #108

**Comment by LTluttmann** [comment]
what do you think of letting the environment classes define the **select_start_nodes** and **get_num_starts** functions?

**Reply by fedebotu** [reply]
I agree with you, this would make it more modular! Thoughts @cbhua ?

PS: maybe we don't need to enforce it for all environments, but add it only to envs that may require this feature (i.e. Implement documentation in `RL4COEnvBase` and raise error if the env did not implement it)
